app/scripts/modules/core/src/search/infrastructure/infrastructureSearch.service.ts

```diff
@@ -99,7 +97,7 @@ export class InfrastructureSearcher {
         formatter = providerFormatter[category];
       }
     }
-    return formatter(entry, fromRoute);
+    return this.$q.when(formatter(entry, fromRoute));
```

Comment:
Is the expectation here that the formatter might not always return a promise? Seems reasonable and nice to allow the flexibility to implementers, just curious if that's the case.

Reply:
None of the formatters returned promises.  Maybe fast properties does?